Office 365/o365/ingest/parser.yml

```
@@ -162,12 +168,71 @@ stages:
           event.action: event.type
         fallback: '["info"]'
       - set:
+          user.id: "{{json_event.message.MailboxOwnerSid}}"
+          office365.exchange.mailbox_guid: "{{json_event.message.MailboxGuid}}"
+          office365.exchange.modified_properties: "{{json_event.message.ModifiedProperties}}"
           office365.context.aad_session_id: "{{json_event.message.SessionId}}"
 
   parse_exchange_item_group:
     actions:
+      - set:
+          event.category: ["email", "file"]
+
+      - translate:
+        dictionary:
+          "Create": ["info", "creation"]
+          "Update": ["info", "change"]
+          "HardDelete": ["info", "deletion"]
+          "SoftDelete": ["info", "deletion"]
+          "MoveToDeletedItems": ["info", "deletion"]
+          "MailItemsAccessed": ["info"]
+          "MailboxLogin": ["start"]
+        mapping:
+          event.action: event.type
+        fallback: ["info"]
+
+      - set:
+          process.name: "{{json_event.message.ClientProcessName}}"
+          office365.exchange.client_version: "{{json_event.message.ClientVersion}}"
+          office365.exchange.email.subjects: >
+            [
+              {% if json_event.message.AffectedItems != null %}
+                {% for item in json_event.message.AffectedItems %}
+                  {{ item.Subject }}
+                {% endfor %}
+              {% endif %}  
+            ]
+          office365.exchange.email.paths: >
+            [
+              {% if json_event.message.AffectedItems != null %}
+                {% for item in json_event.message.AffectedItems %}
+                  {{ item.ParentFolder.Path }}
+                {% endfor %}
+              {% endif %}  
+            ]
+
+          email.attachments: >
+            [
+              {% for email in json_event.message.AffectedItems %}
+                {% for file in email.Attachments.split("; ") %}
+                  {"file":{ "name": "{{ file.split(" (")[0] }}", "size": "{{ file.split(" (")[1].strip('b)') }}" }},
+                {% endfor %}
+              {% endfor %}
+            ]
       - set:
           office365.context.aad_session_id: "{{json_event.message.SessionId}}"
+  parse_share_point:
+    actions:
+      - set:
+          event.category: ["iam"]
+          event.type: ["info"]
```

Sharepoint is more about files and documents. I suggest
```suggestion
          event.category: ["file"]
          event.type: ["access"]
```
instead.